java/spacewalk-java.spec

```diff
@@ -32,14 +32,13 @@
 %define appdir          /srv/tomcat/webapps
 %define jardir          /srv/tomcat/webapps/rhn/WEB-INF/lib
 %define apache_group    www
-%define salt_user-group    salt
+%define salt_user_group salt
 %define apache2         apache2
 %define java_version    11
 %else
 %define appdir          %{_localstatedir}/lib/tomcat/webapps
 %define jardir          %{_localstatedir}/lib/tomcat/webapps/rhn/WEB-INF/lib
 %define apache_group    apache
-%define salt_user_group root 
```

Comment:
@sbluhm we have a special user because tomcat need to write in that directory. Not sure as which user tomcat is running on CentOS.

@Etheryte I think removing is wrong as it it must be defined when building this package for CentOS.

Reply:
tomcat is actually by the user tomcat on CentOS7. Why are you running into the else statement? For SUSE it should stay as `salt`.

1. We could put back the user as `salt`  in line 702 and 703 and just keep the variable salt_user_group as salt_user. We would cover the groups salt (SUSE), tomcat (other).
2. Add tomcat to the group salt. I assume this is being done for SUSE systems?

Reply:
I think I was wrong. tomcat uses salt-api to write in this directory. So it is important under which user salt-api/salt-master is running. With the salt package shipped with Uyuni this is the user "salt".

I would not suggest to use a different salt package, as upstream is slow with merging our new features. Even the latest salt upstream package might miss features which are required for uyuni.

So I think this is now good to be merged. We can make detail changes at any time later if needed.

Reply:
To remove complexity, does it make sense to revert my changes and hardcode the user/group again? Or the least remove the definition out of the if-loop?

Reply:
Personally I think it's fine as-is right now, as good overview of the similarities and differences in the setups similar to apache etc. I'm going to merge this now to see if it resolves the test suite issues fully, and we can come back to this later if it rubs someone the wrong way.